File: Frontend/src/app/shared/models/BuyerOrder.ts
import { Account } from "./Account";
import { SellerOrder } from "./SellerOrder";

export interface BuyerOrder{
    id?: number,
    totalAmount?: number;
    status?: string;
    date?: Date;
    sellerOrders?:SellerOrder[],
    buyer?:Account,
}

File: Backend/MarketplaceBackend-main/src/main/java/com/example/marketplace/model/BuyerOrder.java
package com.example.marketplace.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class BuyerOrder extends Order{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Account buyer;
    @OneToMany(mappedBy = "buyerOrder")
    private List<SellerOrder> sellerOrders;
}
